runtime/src/deterministic.rs

```diff
@@ -227,15 +227,59 @@ pub struct Executor {
     time: Mutex<SystemTime>,
     tasks: Arc<Tasks>,
     sleeping: Mutex<BinaryHeap<Alarm>>,
-    partitions: Mutex<HashMap<String, Partition>>,
     shutdown: Mutex<Stopper>,
     finished: Mutex<bool>,
-    recovered: Mutex<bool>,
 }
 
+impl Executor {
+    /// Clear all tasks.
+    ///
+    /// We don't consume [Executor] because it must remain upgradable for any tasks
+    /// that interact with the runtime during drop.
+    fn stop(&self) {
+        // Mark executor as finished to prevent new tasks
+        *self.finished.lock().unwrap() = true;
+
+        // Drop all wakers
+        self.sleeping.lock().unwrap().clear();
```

Comment:
While not strictly necessary to drop `wakers` prior to clearing tasks, I find it to be good hygine.